src/plugins/intel_gpu/include/intel_gpu/plugin/remote_context.hpp

```diff
@@ -353,8 +353,8 @@ class TypedExecutionContext : public TpublicContextAPI {
 #else
     using surf_key = _Key<cldnn::shared_surface, uint32_t>;
 #endif
-    std::map<surf_key, InferenceEngine::RemoteBlob::Ptr> shared_surf_reg;
-    std::map<cldnn::shared_handle, InferenceEngine::RemoteBlob::Ptr> shared_obj_reg;
+    std::map<surf_key, std::weak_ptr<InferenceEngine::RemoteBlob>> shared_surf_reg;
+    std::map<cldnn::shared_handle, std::weak_ptr<InferenceEngine::RemoteBlob>> shared_obj_reg;
 
```

Comment:
Hi @vladimir-paramuzov , I'm trying the solution you mentioned in (cvs-93914), and this do solve the lifecycle issue with the specific failure described in the jira.
But, there are other failures as in https://github.com/openvinotoolkit/openvino/blob/master/src/tests/functional/plugin/gpu/remote_blob_tests/dx11_remote_ctx_test.cpp#L188
if using weak ptr to the blob, the shared blob will not work in this situation with the above case.
any further insights?

Reply:
does it make sense to unregister these blobs in the context when they are not used?

Reply:
looks like https://github.com/openvinotoolkit/openvino/pull/14244 has already resolved the issue by holding ref to engine? @vladimir-paramuzov